akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1934,7 +1962,7 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
       if (isSuccess && isDelete) DeleteSuccess(key, req)
       else if (isSuccess) UpdateSuccess(key, req)
       else if (isTimeoutOrNotEnoughNodes && isDelete) ReplicationDeleteFailure(key, req)
-      else if (isTimeoutOrNotEnoughNodes) UpdateTimeout(key, req)
+      else if (isTimeoutOrNotEnoughNodes || !durable) UpdateTimeout(key, req)
```

Comment:
I think this was something rather unrelated that I found when testing something. `StoreFailure` should only be used for durable entries.